odoo/orm/domains.py

```
@@ -1409,8 +1403,7 @@ def _optimize_like_str(condition, model):
     if isinstance(value, str):
         return condition
     if isinstance(value, SQL):
-        warnings.warn("Since 19.0, use Domain.custom(to_sql=lambda model, alias, query: SQL(...))", DeprecationWarning)
-        return condition
+        condition._raise("Use Domain.custom instead of SQL", error=TypeError)
```

There are a few places that we could just skip the isinstance and let it crash.